test/integration/nighthawk_test_server.py

```diff
@@ -30,41 +33,85 @@ def __init__(self, server_binary_path, config_template_path, server_ip, ip_versi
     self.server_process = None
     self.server_ip = server_ip
     self.socket_type = socket.AF_INET6 if ip_version == IpVersion.IPV6 else socket.AF_INET
-    self.server_port = -1
-    self.admin_port = -1
     self.admin_address_path = ""
     self.parameterized_config_path = ""
     self.instance_id = str(random.randint(1, 1024 * 1024 * 1024))
     self.parameters = parameters
     self.server_binary_config_path_arg = server_binary_config_path_arg
-
     self.parameters["server_ip"] = self.server_ip
-    with open(self.config_template_path) as f:
-      config = Template(f.read())
-      config = config.substitute(self.parameters)
-      logging.info("Parameterized server configuration: %s", config)
-
-    with tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".yaml") as tmp:
+    self.docker_image = os.getenv("NH_DOCKER_IMAGE", "")
+    self.tmpdir = os.path.join(os.getenv("TMPDIR", "/tmp/nighthawk_benchmark/"), tag + "/")
+    self.parameters["tmpdir"] = self.tmpdir
+    self.parameters["tag"] = tag
+
+    def substitute_yaml_values(obj, params):
```

Comment:
Can we define this as a standalone (global scope) function instead to make the __init__ method shorter and easier to follow?

Once we do, we should also document it.